carbonmark/components/pages/Retire/Activity/Table.tsx

```diff
@@ -0,0 +1,94 @@
+import { cx } from "@emotion/css";
+import { KlimaRetire } from "@klimadao/lib/types/subgraph";
+import { trimWithLocale } from "@klimadao/lib/utils";
+import { t, Trans } from "@lingui/macro";
+import LaunchIcon from "@mui/icons-material/Launch";
+import { Spinner } from "components/shared/Spinner";
+import { Text } from "components/Text";
+import Link from "next/link";
+import { FC } from "react";
+import * as styles from "./styles";
+
+const getFormattedDate = (timestamp: string, locale: string) => {
+  const date = new Date(parseInt(timestamp) * 1000); // expects milliseconds
+  return new Intl.DateTimeFormat(locale, {
+    day: "2-digit",
+    month: "long",
+    year: "numeric",
+  }).format(date);
+};
+
+const createReceiptLink = (address: string, index: string) =>
+  `/retirements/${address}/${Number(index) + 1}`;
+
+type Props = {
+  hasRetirements: boolean;
+  retirements: KlimaRetire[] | false;
+  address: string;
+  locale: string;
+  isLoadingRetirements: boolean;
+  noRetirements: boolean;
+  errorMessage: string;
+  initializing: boolean;
+  className: string;
+};
+
+export const ActivityTable: FC<Props> = (props) => {
+  return (
+    <div className={cx(styles.list, props.className)}>
+      <div className={styles.listHeader}>
+        <Text t="body4">{t`PROJECTS`}</Text>
+        <Text align="end" t="body4">{t`QTY`}</Text>
+        <Text t="body4">{t`DATE`}</Text>
+      </div>
+
+      {props.retirements &&
+        props.retirements.map((r) => (
+          <Link
+            key={r.id}
+            href={createReceiptLink(props.address, r.index)}
+            target="_blank"
+          >
+            <div className={styles.listItem}>
+              <Text>{r.offset.name || r.offset.projectID}</Text>
```

Comment:
we are now running a carbon-projects CMS that has normalized names (with more consistent capitalization and punctuation). Would be preferable to use those names. Unfortunately that adds some complexity around merging in the names w/ the retirements array.

We can query the CMS directly from the client for v1, that's no problem, but we can also think about an `/retirements` endpoint which would handle caching and reduce the cms costs. Perhaps you find it easier to just write a rudimentary `pages/api/retirements` endpoint right now to keep the react component simple?

Reply:
Done